app/components/book-list.ts

```diff
@@ -46,35 +69,15 @@ export default class BookListComponent extends Component<BookListSignature> {
     return options;
   }
 
-  @cached
-  get currentPage() {
-    const _firstPageOptions = this._firstPageOptions;
-    const firstPageOptions = this.firstPageOptions;
-    const currentUrl = this.currentUrl;
-
-    // if the first page options changed, we need to fetch a new first page
-    if (_firstPageOptions?.url !== firstPageOptions.url) {
-      return this.fetchPage(firstPageOptions);
-    }
-
-    return this.fetchPage(currentUrl ? { url: currentUrl } : firstPageOptions);
-  }
-
-  get books(): Collection<Book> | null {
-    return this.currentPage.content || null;
+  get currentBooks(): Collection<Book> | null {
+    return this.currentPage.current.content || null;
   }
 
-  fetchPage(options: { url: string }) {
-    const dataWrapper = this.dataWrapper;
-    const future = this.store.request<Book>(options);
-
-    future.then((books) => {
-      dataWrapper.content = books.content;
-      this.links.addPage(books.content);
-    });
-
-    return dataWrapper;
-  }
+  // absorbs null-flashes as underlying data changes
```

Comment:
null-flashes only occur because we create a new state context each time tracked data changes: https://github.com/runspired/ember-data-collections-demo/pull/1/files#diff-64e5910d87bd4744480041f7b84746de88b9f83fa6fddbfbd1127d4a51a8b391R54


we could pull that up in to the outer function and avoid that problem tho, and then we wouldn't need keeplatest